### ISSUE TITLE
Loss of network connection during 'mintupgrade uprade' causes troubles

### ISSUE THREAD

**Issue by szeder** [linuxmint/mintupgrade-legacy#8]
Here is what happened while upgrading from 17.3 to 18 Cinnamon, following the directions in https://community.linuxmint.com/tutorial/view/2316:
- `mintupgrade check` was successful on the first run.
- `mintupgrade download` was successful on the first run, too.  According to the docs linked above, this should "_download the packages necessary to upgrade to Linux Mint 18_".  As it turns out, it doesn't.
- During the "_Performing upgrade_" step of `mintupgrade upgrade` my wifi went down for whatever reason.  It came back shortly on all other devices, but not on the computer I was upgrading: the Network Manager applet listed the SSID, tried to establish connection, but it couldn't.  Oh, well, I presume a lot of things changed behind the back of the running NetworkManager process, so I think that's unfortunate but understandable.  No worries, because all packages have already been downloaded earlier.  Right?
- Wrong, because then came the "_Re-installing the meta-package for your edition of Linux Mint_" step, and `mintupgrade` tried to download some additional packages, which failed, of course, leaving my computer in a partially upgraded state with no internet connection.  Uh, oh.
- For lack of a better idea I rebooted, fearing that I won't have a working desktop environment but hoping that NM will find a connection and I will ba able to sort out the missing packages from a console.  And that's exactly what happened: after reboot I was greeted with a black screen with a mouse pointer in the middle, but I got network connection, and luckily `mintupgrade upgrade` was able to continue where it left off, finishing the upgrade.  Phew.

I think `mintupgrade download` should download **all** necessary packages, including those needed only in the later "_Re-installing the meta-package for your edition of Linux Mint_", "_Re-installing the multimedia codecs_", and "_Installing new packages_" steps.

**Comment by thomaspaulb** [comment]
I had this problem also. I rebooted and was left in the same state of network-less black screen... The steps I needed to follow in order to fix:

1. Connect a cable to my laptop, configure `eth3` in `/etc/network/interfaces` and run `ifup eth3`
2. Clone this `mintupgrade` repository (mintupgrade was removed in my Sarah somehow)
3. Copy the supporting files into `/usr/share/linuxmint/mintupgrade`
4. Change the mintupgrade script to remove `mintupgrade` [here](https://github.com/linuxmint/mintupgrade/blob/master/usr/bin/mintupgrade#L16).
5. Rerun `mintupgrade upgrade`
6. Run `sudo apt-get install xfonts-100dpi xfonts-cyrillic`

I am still left with the problem that my X server does not start automatically, I have to actively run `startx`. My X log file shows a segfault. @clefebvre tips? Will keep you in mind during Christmas donations : D
